app/views/spree/addresses/_form.html.erb

```diff
@@ -1,32 +1,134 @@
+          <%= address_form.label :address2 do %>
+            <%= Spree.t(:address2) %>
+          <% end %>
+        </div>
+      </div>
+    <%elsif field == "country" %>
+      <div class="col-sm-6 " id="<%= "#{address_id}country" %>">
+        <div class="form-group form-floating mb-20" id="<%= "#{address_id}country-selection" %>" >
+          <%= address_form.collection_select :country_id, available_countries, :id, :name,
+                                              { prompt: Spree.t(:country).upcase },
+                                              { class: 'custom-select',
+                                                aria: { label: Spree.t(:country) } } %>
+          <%= address_form.label(Spree.t(:country)) %>
+        </div>
+      </div>
+    <%elsif field == "state" %>
+      <div class="col-sm-6 " >
+        <%= address_field(address_form, :state, address_id) { address_state(address_form, address.country, address_id) }  %>
+      </div>
+    <%elsif field == "city" %>
+      <div class="col-sm-6 ">
+        <div class="form-group form-floating mb-20">
+          <%= address_form.text_field :city, class: 'form-control', placeholder:"City*", required: true %>
+          <%= address_form.label :city do %>
+            <%= Spree.t(:city) %><abbr>*</abbr>
+          <% end %>
+        </div>
+      </div>
+    <%elsif field == "zipcode" %>
+        <div class="col-sm-6 ">
+        <%= address_field(address_form, :zipcode, address_id) { address_zipcode(address_form, address.country, address_id) } %>
+      </div>
+    <%elsif field == "phone" %>
+      <div class="col-sm-12">
+        <div class="form-group form-floating mb-10">
+          <%= address_form.text_field :phone, class: 'form-control', placeholder:"Phone Number*", required: true %>
+          <%= address_form.label :phone do %>
+            <%= Spree.t(:phone) %><abbr>*</abbr>
+          <% end %>
+        </div>
+      </div>
+    <% else %>
+      <div class="col-sm-6 ">
+        <%= address_field(address_form, field.to_sym, address_id) %>
+      </div>  
+    <% end %>
+  <% end %>
+  <% if spree_current_user.present? %>
+    <div class="col-sm-12">
+      <div class="custom-checkbox-wrap d-flex flex-wrap align-items-center mb-60">
+        <% is_user_first_address = current_spree_user.addresses.present? ? true : false %>
+        <%if( request.fullpath.include? "/checkout")%>
+          <div class="custom-checkbox <%#= is_user_first_address ? '' : 'd-none'%>" >
+            <label>
+              <input type="checkbox" class="custom-control-input" checked>
+              <span class="custom-control-label"></span>Save to My Addresses</label>
+            <label>
+          </div>
+        <% end %>
+        <div class="custom-checkbox">
+            <%= address_form.check_box :is_default, class: 'custom-control-input', checked: is_user_first_address ? "" : "checked" %>
```

Comment:
checked condition is not good. It should check the default address presence like this.
user.addresses.default